dotnet-runtime-90/snap/snapcraft.yaml

```diff
@@ -0,0 +1,84 @@
+name: dotnet-runtime-90
+base: core22
+adopt-info: dotnet-runtime
+summary: .NET 9.0 Runtime.
+description: |
+  .NET Core runtimes and libraries which are optimized for running .NET Core apps in production. See https://dot.net/core.
+  .NET Core is a general purpose development platform maintained by Microsoft.
+
+grade: stable
+confinement: strict
+
+package-repositories:
+ - type: apt
+   ppa: dotnet/snaps
+   priority: always
+
+architectures:
+  - build-on: [amd64]
+    build-for: [amd64]
+  - build-on: [arm64]
+    build-for: [arm64]
+
+slots:
+  dotnet-runtime:
+    content: dotnet-runtime-90
+    interface: content
+    source:
+      read:
+        - $SNAP/usr/lib/dotnet
+
+parts:
+  dotnet-runtime:
+    plugin: nil
+    source: .
+    stage-packages:
+      - dotnet-runtime-9.0
+    override-stage: |
+      craftctl default
+
+      if [ "${CRAFT_ARCH_BUILD_FOR}" = "amd64" ]; then
+        DOTNET_RUNTIME_VERSION=$(LD_LIBRARY_PATH=${CRAFT_STAGE}/usr/lib/x86_64-linux-gnu ${CRAFT_STAGE}/usr/lib/dotnet/dotnet --list-runtimes \
+          grep NETCoreApp | head -n 1 | awk '{print $2}')
+      elif [ "${CRAFT_ARCH_BUILD_FOR}" = "arm64" ]; then
+        DOTNET_RUNTIME_VERSION=$(LD_LIBRARY_PATH=${CRAFT_STAGE}/usr/lib/aarch64-linux-gnu ${CRAFT_STAGE}/usr/lib/dotnet/dotnet --list-runtimes \
+          grep NETCoreApp | head -n 1 | awk '{print $2}')
+      else
+        echo "Unknown architecture (${CRAFT_ARCH_BUILD_FOR})"
+        exit 1
+      fi
+
+      if [ -f ${CRAFT_PART_SRC}/GIT_SHA ]; then
+        craftctl set version="${DOTNET_RUNTIME_VERSION}+git.$(cat ${CRAFT_PART_SRC}/GIT_SHA)"
+      else
+        craftctl set version="${DOTNET_RUNTIME_VERSION}"
+      fi
+
```

Comment:
**note:** The .NET 6, 7 & 8 runtime `snapcraft.yaml` files have an `override-prime` section here:

```
    override-prime: |
      # This is for backwards compatibility:
      # Up until now, .NET Runtime content snaps would place the .NET directory hive
      # in /. As the binaries are now pulled from Canonical's deb package, which installs
      # the hive into /usr/lib/dotnet, we place a `dotnet` symlink in / so that it's
      # still where people might expect it to be.
      ln --symbolic usr/lib/dotnet/dotnet "${CRAFT_PRIME}"/dotnet
      craftctl default
```

I assume that we do not need this for .NET 9, because it is a new snap and there is no one whose expectations we are breaking.

**question(non-blocking):** Who might actually use the symlink in the 6,7,8 snaps? / Do you think there is someone whose use case we are breaking when we would remove that from the other runtime `snapcraft.yaml` files?

Reply:
Your assumption is correct 🙂 

I have seen cases of people using the content snaps before we made the change described in the `override-prime` comment up there, e.g. canonical/dotnet-snap#14, which is what motivated me to create the symlink in the first place.